From 9a2922e1ccb326f03828deaa66dfb8bd0b1d5854 Mon Sep 17 00:00:00 2001
From: Martin Hrdlicka <martin.hrdlicka@nxp.com>
Date: Thu, 2 Mar 2023 16:19:24 +0100
Subject: [PATCH 61/63] arm64: dts: s32g-pfe: add remote HIF netdevs

Additional netdevs, one for Master and one
for Slave are introduced.

Issue: AAVB-6542

Upstream-Status: Pending 

Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
Signed-off-by: Martin Hrdlicka <martin.hrdlicka@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi | 12 ++++++++++++
 arch/arm64/boot/dts/freescale/s32g-pfe.dtsi       | 11 +++++++++++
 2 files changed, 23 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi b/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
index e40e24c55e0e..a2bc797dd0a5 100644
--- a/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
@@ -7,6 +7,7 @@
  *   pfe1sl: to EMAC_1
  *   pfe2sl: to EMAC_2
  *   aux0sl: to CLASS
+ *   hif0sl: to HIF_0
  */
 
 #include <dt-bindings/clock/s32g-scmi-clock.h>
@@ -77,6 +78,17 @@ pfesl_aux0: ethernet@103 {
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_3>;
 				nxp,pfeng-netif-mode-aux;
 			};
+
+			/* Network interface 'hif0sl' */
+			pfesl_hif0: ethernet@104 {
+				compatible = "nxp,s32g-pfe-netif";
+				status = "disabled";
+				reg = <104>;
+				local-mac-address = [ 00 04 9F BE FF F0 ];
+				nxp,pfeng-if-name = "hif0sl";
+				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_3>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_HIF_0>;
+			};
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi b/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
index 14fdb47f4828..85a79ab36b65 100644
--- a/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
@@ -161,6 +161,17 @@ pfe_aux0: ethernet@13 {
 							 <PFE_HIF_CHANNEL_2>;
 				nxp,pfeng-netif-mode-aux;
 			};
+
+			/* Network interface 'hif3' */
+			pfe_hif3: ethernet@14 {
+				compatible = "nxp,s32g-pfe-netif";
+				status = "disabled";
+				reg = <14>;
+				local-mac-address = [ 00 04 9F BE EF F0 ];
+				nxp,pfeng-if-name = "hif3";
+				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_0>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_HIF_3>;
+			};
 		};
 	};
 };
-- 
2.25.1

